apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,180 @@
+	//
+	// fooRef:
+	//   name: myservice
+	//
+	// Otherwise, if the services resource is not desired, specify the desired
+	// group:
+	//
+	// fooRef:
+	//   group: acme.io
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +kubebuilder:default=services
+	// +kubebuilder:validation:MaxLength=253
+	Resource string `json:"resource,omitempty"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+
+	// Port is the port of the referent. If unspecified, this policy applies to
+	// all ports on the backend.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	Port *int32 `json:"port,omitempty"`
+}
+
+// BackendTLSConfig describes TLS configuration for a backend.
+type BackendTLSConfig struct {
+	// ClientCertificateRef is a reference to a TLS client certificate-key pair
+	// that may be used to connect to these backends. If an entry in this list
+	// omits or specifies the empty string for both the group and the resource,
+	// the resource defaults to "secrets". An implementation may support other
+	// resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
+	//
+	// If a Secret is referenced, it must be of type "kubernetes.io/tls" and
+	// contain tls.crt and tls.key data fields that contain the certificate and
+	// private key to use for TLS.
+	//
+	// Support: Extended
+	//
+	// +optional
+	ClientCertificateRef *CertificateObjectReference `json:"clientCertificateRef,omitempty"`
+
+	// CertificateAuthorityRef is a reference to a resource that includes
+	// trusted CA certificates for the associated backends. If an entry in this
+	// list omits or specifies the empty string for both the group and the
+	// resource, the resource defaults to "secrets". An implementation may
+	// support other resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
+	//
+	// When stored in a Secret, certificates must be PEM encoded and specified
+	// within the "ca.crt" data field of the Secret. Multiple certificates can
+	// be specified, concatenated by new lines.
+	//
+	// Support: Extended
+	//
+	// +optional
+	CertificateAuthorityRef *CertificateObjectReference `json:"certificateAuthorityRef,omitempty"`
+
+	// Options are a list of key/value pairs to give extended options to the
+	// provider.
+	//
+	// Support: Implementation-specific.
+	// +optional
+	Options map[string]string `json:"options,omitempty"`
+}
+
+// BackendPolicyStatus defines the observed state of BackendPolicy. Conditions
+// that are related to a specific Route or Gateway should be placed on the
+// Route(s) using backends configured by this BackendPolicy.
+type BackendPolicyStatus struct {
+	// Conditions describe the current conditions of the BackendPolicy.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
```

Comment:
Should we have the limit of 5 conditions like we did in the API cleanup PR?

Reply:
I think I'd rather save that for the API cleanup PR so we can have consistent limits everywhere.